backend/inspirehep/disambiguation/tasks.py

```diff
@@ -114,19 +114,28 @@ def assign_reference_to_author_if_unambiguous_literature_author_match(
     )
     if matched_ref:
         author_ids = get_value(
-            matched_record_list[0], "inner_hits.authors.hits.hits._source.ids", []
+            matched_record_list[0], "inner_hits.authors.hits.hits[0]._source.ids", []
         )
         assign_bai_to_author(author_ids, author)
         return matched_ref
 
 
-def assign_bai_to_author(author_ids, author):
-    author_bai_list = get_values_for_schema(author_ids, "INSPIRE BAI")
-    if author_bai_list and "INSPIRE BAI" not in get_value(author, "ids.schema", []):
-        author["ids"] = [
-            *author.get("ids", []),
-            {"schema": "INSPIRE BAI", "value": author_bai_list[0]},
-        ]
+def assign_bai_to_author(matched_author_ids, literature_author):
+    matched_author_bai_list = get_values_for_schema(matched_author_ids, "INSPIRE BAI")
+    if not matched_author_bai_list:
+        return
+    literature_author["ids"] = [
+        *literature_author.get("ids", []),
+        {"schema": "INSPIRE BAI", "value": matched_author_bai_list[0]},
+    ]
+
+
+def get_literature_author_bai(literature_author):
+    literature_author_ids = literature_author.get("ids", [])
+    for idx, author_id_dict in enumerate(literature_author.get("ids", [])):
```

Comment:
I'm not sure I understand this code, why couldn't we use https://github.com/inspirehep/inspire-utils/blob/master/inspire_utils/record.py#L94?

```python
get_values_for_schema(literature_author, 'INSPIRE BAI')
```

Reply:
The case is that I would like to remove the whole `{"schema": "INSPIRE BAI", "value" "A.Value"}` from the id list (because it can't be used later for matching) and `get_values_for_schema` is returning only the value